### ISSUE TITLE
Request: show Arch package news stuff when updating

### ISSUE THREAD

**Issue by samvdkris** [Jguer/yay#384]
I saw a comment on Reddit about someone who made an alias to run `curl -s https://www.archlinux.org/feeds/news/ | xmllint --xpath //item/title\ \|\ //item/pubDate /dev/stdin | sed -r -e "s:<title>([^<]*?)</title><pubDate>([^<]*?)</pubDate>:\2\t\1\n:g"` before running `yay -Syu` to see if there are any packages that need manual intervention or have any problems. I thought it would be cool if yay automatically did this when updating. I'd make a pull request, but I know absolutely nothing about Go and even making a script that just executed `ls` failed for some reason. If anyone could add this (or another implementation of something similar), that would be really cool.

Also, I just wanted to say this AUR helper is really cool. Thanks to everyone who contributed to it! :)

**Comment by heyitswither** [comment]
Probably best to add this as an option, maybe `--news`

**Comment by samvdkris** [comment]
Yeah, I guess some people would just get annoyed. Or maybe just make it togglable in `~/.conf/yay/config.json` (already exists for other settings).

**Comment by heyitswither** [comment]
Since you brought up the config file, it might be cool to add "hooks," commands that will be run on certain events. So instead of hard-coding this feed command, users can put whatever they want to run on updates.

**Comment by Morganamilo** [comment]
On mobile wrong button.

I do want to get some comment and news fetching into yay, probably as an extension to -P.

No plant to implement right now but I like the idea.

**Comment by PeachIceTea** [comment]
-P seems perfect for that and --news would work as well. However -Pn is already taken for a different function, maybe -Pw would do?

**Comment by Morganamilo** [comment]
I've thought about -W/--web also, this might deserve its own namespace.

**Comment by PeachIceTea** [comment]
It does seem like something that people might wanna access more quickly. Also it would be nice because the short and the long form would match. It is however a bit ambiguous.

**Comment by Morganamilo** [comment]
What makes it ambiguous?

**Comment by PeachIceTea** [comment]
I just think that it is not as obvious as others, e.g. --sync or --upgrade. --web isn't immediately clear in what it does.

**Comment by ashleybone** [comment]
An alternative to adding this feature is using pacmatic, a pacman wrapper in the community repo that does this (it also prints mailing list chatter).  pacmatic has an environmental variable, pacman_program, that lets you specify an alternative program to wrap.  I've been using it with yay for a while.

**Comment by Morganamilo** [comment]
News is already added via `yay -Pw`. It's not shown during `-Syu` currently though, you have to explicitly ask for it.

Stuff like mailing list chatter and such is definitely out of scope for Yay. Pacmatic would definitely be the recommended solution if anybody wants anything more than news.

**Comment by HaleTom** [comment]
As @Morganamilo mentioned `pacmatic`, here's my solution:

[Have `pacmatic` wrap `yay` wrap `powerpill` wrap `pacman`
](https://unix.stackexchange.com/a/384102/143394)

**Comment by Jpfonseca** [comment]
> News is already added via `yay -Pw`. It's not shown during `-Syu` currently though, you have to explicitly ask for it.


I was trying this, but was unable to fetch any...

**Comment by antonovr** [comment]
@Jpfonseca 
according man 
`       -w, --news
              Print new news from the Archlinux homepage. News is considered new if it is newer than the build date of all native packages. Pass this twice to show all available news.
`
so `yay -Pww` will show all news

**Comment by AlexWayfer** [comment]
This is actual, I've faced with https://www.archlinux.org/news/nvidia-45528-is-incompatible-with-linux-59/ a few days ago.

**Comment by AlexWayfer** [comment]
Oh, stale policy became harder…

**Comment by grandchild** [comment]
@AlexWayfer What's unresolved about this ticket? Just run `yay -Pw ; yay -Syu` and you're golden. `yay -Pww` if you're unsure.

**Comment by AlexWayfer** [comment]
> @AlexWayfer What's unresolved about this ticket? Just run `yay -Pw ; yay -Syu` and you're golden. `yay -Pww` if you're unsure.

Hm, let me check please. How to read help for `-P` option, for example for `-w` sub-options?

```
> yay -P --help
pacman: invalid option -- 'P'
~
> yay --show --help
pacman: unrecognized option '--show'
```

**Comment by grandchild** [comment]
`man yay`? Search for `--news`.

I generally use `yay -Pwwq`, that way I only get the headlines, but don't have to worry about missing anything because I ran an update without printing news.

**Comment by AlexWayfer** [comment]
> `man yay`? Search for `--news`.
> 
> I generally use `yay -Pwwq`, that way I only get the headlines, but don't have to worry about missing anything because I ran an update without printing news.

OK, thank you, it helped, but I still don't understand what the second `-w` does, I found no information in `man`.

---

So, about this issue: notice "when updating". These two commands is kind of work-around, but the issue is about automatic displaying I think.

So, you run `yay` (or explicitly `yay -Syu`) and you see Arch news as well as outdated, packages orphaned and other info.

Also it could be handful with getting/displaying news only after the last upgrade, so relevant and not too old, or by date (last month), I'm not sure. As I understand, we can't filter them by installed packages, because news have no such meta-information and parsing can be incorrect.

**Comment by grandchild** [comment]
> -w, --news
>   Print new news from the Archlinux homepage. News is considered new if it is newer than the build date of all native packages. **Pass this twice to show all available news.**

Reads clearly to me.

I don't think that using two commands is a workaround. This is what the Unix shell is for. Composing commands is great. Not everything has to be baked into every command. Mix and match! 🙂 You could even build a little prompt wait in between:
```
yay -Pwwq ; echo -n "All good?" ; read ; yay -Syu
```
Put all that into an alias, and you're good to go. Make your own tools!

---

> Also it could be handful *["helpful" I assume?]* with getting/displaying news only after the last upgrade

But that is exactly what the single `-w` does no?

> filter them by installed packages

The frequency of news posts is so low, that on any given update you're bound to get only one news item if any. That's not hard to filter and see if it's relevant in my opinion. And as you say it would be hard to parse the news for that (sometimes it's not even related to any package and at the same time all, like the man-page service recently). So the *effort* that would have to be put into the code looking for that is so astronomically higher than the usefulness, that I'd be strongly against any attempt to implement it.

**Comment by AlexWayfer** [comment]
> > -w, --news
> > Print new news from the Archlinux homepage. News is considered new if it is newer than the build date of all native packages. **Pass this twice to show all available news.**
> 
> Reads clearly to me.

Oh, sorry, didn't notice, not familiar format for me. It even seems like resolving "show only actual news", described by me, although not so obvious (what is "native packages"?).



> I don't think that using two commands is a workaround.

I think, it should decide maintainers (@Morganamilo) and author of this issue (@samvdkris), is it acceptable solution or just work-around. I assume that this is a work-around from the issue title and my personal wishes.



> This is what the Unix shell is for. Composing commands is great. Not everything has to be baked into every command. Mix and match!  You could even build a little prompt wait in between:
> 
> ```
> yay -Pwwq ; echo -n "All good?" ; read ; yay -Syu
> ```
> 
> Put all that into an alias, and you're good to go. Make your own tools!

I can agree that composing commands is great, but…

1. It (news in updating command) is still more handful, especially for newbies, than manual composing by every interested user, and somebody (as I some time ago) can even don't know about news and package migrations, and faced with it only after problems.
2. I have to choose a name for alias (or even sub-program), and if I want to have the same as a program (`yay`), for consistence (`yay` now has `-Syu` as default at all) — I'd lose common program usage, so I have to choose something custom like `yay-pw-syu` or `yay-upgrade`.
3. Once more: there is already an information about out-dated packages, orphaned packages, deleted packages — why we can't to add info from news, especially when we already have an interface to get them? Or should we take all of this information out for commands composing?



> ["helpful" I assume?]

Yes, sorry, these words seem like synonyms for me. Sorry for mistake.



> > Also it could be handful _["helpful" I assume?]_ with getting/displaying news only after the last upgrade
> 
> But that is exactly what the single `-w` does no?

Yes, as I wrote above. I just didn't know what the second `-w` does, as I wrote in the previous comment.



> > filter them by installed packages
> 
> The frequency of news posts is so low, that on any given update you're bound to get only one news item if any. That's not hard to filter and see if it's relevant in my opinion. And as you say it would be hard to parse the news for that (sometimes it's not even related to any package and at the same time all, like the man-page service recently). So the _effort_ that would have to be put into the code looking for that is so astronomically higher than the usefulness, that I'd be strongly against any attempt to implement it.

That's why I wrote "we can't filter them by installed packages" ("**we can't**"), please, don't take my words out of a context. 🙃

**Comment by grandchild** [comment]
> That's why I wrote "we can't filter them by installed packages" ("we can't"), please, don't take my words out of a context.

Sorry, I apologise, didn't mean to be rude!

**Comment by AlexWayfer** [comment]
> > That's why I wrote "we can't filter them by installed packages" ("we can't"), please, don't take my words out of a context.
> 
> Sorry, I apologise, didn't mean to be rude!

OK, I just saw a lot of text about what I didn't mean, like I agree that we should **not** recognize any (installed) packages in news, while there is no meta-information about them and a small overall number of news, especially with double `-w` flag.
